### DISCUSSION TITLE
I'd advice **No Path Based Imports**

### DISCUSSION THREAD

**Discussion by xialvjun** [gleam-lang/gleam#2136] [Ideas & suggestions]
Have a look at https://github.com/koka-lang/koka/issues/333

With **path based imports**, a library can not split items just public in the library from items public to the whole project.

We can have a easy and simple mode:

```
// libA/src/a.gleam
pub const a1 = 123;
const a2 = 234;

// libA/src/b.gleam
pub const b1 = 123;
const b2 = 234;

// libA/src/lib.gleam - All package starts from lib.gleam or main.gleam
// You must define the module, or other it didn't even exist
pub mod a;  // since mod a is pub, so project using this lib can access libA.a.a1
mod b;         // but b and b.b1 is just public in this library

```

**Comment by lpil** [comment]
I'm sorry @xialvjun , I don't understand. Could you explain in more detail please? Thank you

**Reply by xialvjun** [reply]
If you are a library maker (not a app maker), you write some util functions in a module and use then in your library, but you don't want to share them to your library users because semantic version. What can you do?

**Reply by lpil** [reply]
Our current recommendation is to put them under `yourapp/internal`, which will be hidden from users. This can be configured in `gleam.toml` also. https://gleam.run/writing-gleam/gleam-toml/

**Reply by xialvjun** [reply]
Even though it create a new concept, it did solved problems.

**Reply by prettymofas** [reply]
Thanks u>)!
